client/components/phone-input/index.jsx

```diff
@@ -39,6 +42,12 @@ const PhoneInput = React.createClass( {
 		};
 	},
 
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.value !== nextProps.value && nextProps.value ) {
```

Comment:
This checks whether `nextProps.value` is different than `this.props.value` and `nextProps.value` is truthy. 

afaik `nextProps` will always be an object so there's no need to check for `nextProps`.